### ISSUE TITLE
tabs' status issue when discarding them

### ISSUE THREAD

**Issue by NightSharp** [piroor/treestyletab#2530]
I set the tab labels' style to make me get the status of tabs easily.
And when I use TabUnloader for TST or Auto Tab Discard to unload tab not discarded
It always happen that:
The tab status first turns to "discared" from "loaded but unread" or "loaded and read"--It looks like normal.
But immediately it turns to "loaded and unread" from "discarded"--Actually the tab has been discarded correctly.
So I discard the tab with the two addons above again.
This time the tab satus turns to "discarded and unread", though It may have already been read before the first discarding.

Sometimes the addon UnloadTabs can work normally, but it is not valid everytime yet.
I hope the tab status can be changed correctly

And another issue is, when some tabs are being discared at the same time, sometimes the focus might be displayed incorrectly or these tabs fail to be dicarded and can not be discared again until I restart Firefox

**Comment by piroor** [comment]
The situation you wrote looks complicated, so it is hard to be investigated by me. Could you marshal "steps to reproduce", "actual result" and "expected result", based on the [template](https://github.com/piroor/treestyletab/blob/master/ISSUE_TEMPLATE.md)? For example:

```
Steps to reproduce:

1. Start Firefox with a clean profile.
2. Configure Firefox to restore tabs on the next startup.
3. Install TST 3.4.7 and TabUnloader for TST.
4. Open a tab with the URL `http://example.com/`.
5. Restart Firefox.

Actual result:
The restored tab with the URL `http://example.com/`doesn't have a `discarded` class.

Expected result:
The restored tab with the URL `http://example.com/` should have a `discarded` class.
```

**Comment by NightSharp** [comment]
@piroor  Well, If I just install TST and TabUnloader for TST, It seems that the issue do not exist at all...
Maybe it is caused by some addons or some changes in about:config, but I don't know how to find them quickly.
when the issue exists, steps to reporduce:
1. set different styles for diffrent status of tabs (in order to see their status quickly).
2. open a page containing some URLs. (loaded and read)
3. open some new pages from the parent above in background. (loaded and unread)
4. use Tab Unloader for TST to discard any one of them, say "TabA".

Actual result:
1. "TabA" turns to "discarded" at first, but immediately turn away to "loaded and unread" (may be less than 100ms). And in fact "TabA" has already been discarded.
2. I discard it again and this time it turns to "unloaded and unread".

Expected result:
"TabA" turns to "discarded" and keep up the status until it gets focused.



> 
> 
> The situation you wrote looks complicated, so it is hard to be investigated by me. Could you marshal "steps to reproduce", "actual result" and "expected result", based on the [template](https://github.com/piroor/treestyletab/blob/master/ISSUE_TEMPLATE.md)? For example:
> 
> ```
> Steps to reproduce:
> 
> 1. Start Firefox with a clean profile.
> 2. Configure Firefox to restore tabs on the next startup.
> 3. Install TST 3.4.7 and TabUnloader for TST.
> 4. Open a tab with the URL `http://example.com/`.
> 5. Restart Firefox.
> 
> Actual result:
> The restored tab with the URL `http://example.com/`doesn't have a `discarded` class.
> 
> Expected result:
> The restored tab with the URL `http://example.com/` should have a `discarded` class.
> ```

**Comment by NightSharp** [comment]
Well, If I just install TST and TabUnloader for TST, It seems that the issue do not exist at all...

Maybe it is caused by some addons or some changes in about:config, but I don't know how to find them quickly.

when the issue exists, steps to reporduce:

1. set different styles for diffrent status of tabs (in order to see their status quickly).

2. open a page containing some URLs. (loaded and read)

3. open some new pages from the parent above in background. (loaded and unread)

4. use Tab Unloader for TST to discard any one of them, say "TabA".




Actual result:

1. "TabA" turns to "discarded" at first, but immediately turn away to "loaded and unread" (may be less than 100ms). And in fact "TabA" has already been discarded.

2. I discard it again and this time it turns to "unloaded and unread".




Expected result:

"TabA" turns to "discarded" and keep up the status until it gets focused.
















At 2020-03-24 13:59:12, "piroor" <notifications@github.com> wrote:

The situation you wrote looks complicated, so it is hard to be investigated by me. Could you marshal "steps to reproduce", "actual result" and "expected result", based on the template? For example:

Steps to reproduce:

1. Start Firefox with a clean profile.
2. Configure Firefox to restore tabs on the next startup.
3. Install TST 3.4.7 and TabUnloader for TST.
4. Open a tab with the URL `http://example.com/`.
5. Restart Firefox.

Actual result:
The restored tab with the URL `http://example.com/`doesn't have a `discarded` class.

Expected result:
The restored tab with the URL `http://example.com/` should have a `discarded` class.

**Comment by piroor** [comment]
There are two reports #2534 and #2530 for one topic. I hope to close one of them for a simplification. Can I close this?

**Comment by NightSharp** [comment]
Oh, I see. Now I've closed #2530  
















At 2020-03-27 17:36:46, "piroor" <notifications@github.com> wrote:

There are two reports #2534 and #2530 for one topic. I hope to close one of them for a simplification. Can I close this?
